### PR TITLE
fix generics templates

### PR DESCRIPTION
**Pull request by bendavies**
This PR fixes the templates for `MicroMapperInterface` and `MapperInterface`.
regression tests have been added.

as an aside, is the `$toClass` param in `MapperInterface::load(...)` redundant?
It seems we will only ever have concrete implementations here and will just new up a new class, which will match the class provided in the `AsMapper` attribute.

### CODE REVIEW

File: src/MapperInterface.php [hunk 1]
```diff
@@ -39,6 +39,9 @@ public function load(object $from, string $toClass, array $context): object;
      *
      * Receives the "to object" returned from load().
      *
+     * @template TFrom of object
```
Comment (stof) [inline src/MapperInterface.php]: TFrom could be removed entirely. As it is only used in the param which is the source for inferring it, it doesn't provide any benefit over the object type

File: src/MicroMapperInterface.php [hunk 2]
```diff
@@ -11,15 +11,15 @@
...
- *
- * @template TFrom of object
- * @template TTo of object
  */
 interface MicroMapperInterface
 {
     public const MAX_DEPTH = 'max_depth';
 
     /**
+     * @template TFrom of object
```
Comment (stof) [inline src/MicroMapperInterface.php]: Same here

File: src/MapperInterface.php (hunk 1 above)
Comment (bendavies) [reply src/MapperInterface.php]: done. removed `TFrom` everywhere

**Comment by bendavies** [comment]
@weaverryan i don't think these are my failures

**Comment by weaverryan** [comment]
Hmm, actually, I think they are... it's related to deprecations in PHPStan - so my guess is from the new test. But these are "Other deprecation notices"... are those direct? Indirect? I'm actually not sure - we might need to tweak our `SYMFONY_DEPRECATIONS_HELPER`... or figure out the underlying cause of this.

**Comment by stof** [comment]
I guess they are not classified properly because the deprecation is triggered from a class defined inside a phar, so it path has starts with `phar://`, which means it does not start with the absolute path of the vendor folder. I don't find it unexpected that we did not though of the case of phars in the vendor folder in that detection logic.

**Comment by weaverryan** [comment]
I wonder if changing the phpstan dependency to the stable 1.10 in composer.json would help (I can't remember why I have 1.11-dev - probably by accident).

**Comment by bendavies** [comment]
doesn't help. but i tried using phpunit directly instead of (not-so-)simple-phpunit, and there are no deprecations.

**Comment by weaverryan** [comment]
> doesn't help. but i tried using phpunit directly instead of (not-so-)simple-phpunit, and there are no deprecations

Yes, but probably just because it's hiding the deprecations :p. Maybe we can just generate and point to a baseline file for these deprecations - https://symfony.com/doc/current/components/phpunit_bridge.html#baseline-deprecations

**Comment by bendavies** [comment]
done

**Comment by weaverryan** [comment]
Still not happy - not sure why - phpunit-bridge can be pesky to work with and configure. BUT, I have changed the settings so that if you push, the CI will run automatically vs waiting for my approval 🙃

**Comment by bendavies** [comment]
oh yeah, it's hard coded paths in the allowed deprations

**Comment by bendavies** [comment]
can't see the issue atm. there are deprecations in ci that i don't get locally.

**Comment by bendavies** [comment]
hallelujah

**Comment by weaverryan** [comment]
Thank you for your hard work on this Ben!

**Pull request merged**